pkg/util/object-identifier.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2024 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/kubestellar/kubestellar/api/control/v1alpha1"
+)
+
+// ObjectIdentifier struct is used to add items to the workqueue.
+// The ObjectIdentifier contains all the necessary information to retrieve an object.
+type ObjectIdentifier struct {
+	GVK        schema.GroupVersionKind
+	Resource   string
+	ObjectName cache.ObjectName
+}
```

Comment:
This has the same contents as the type SourceRef in the same package. Do we really need two types?

Reply:
I think at the moment we can live with that - soon (my) focus will move to the status controller, which will involve redesigning some things, and aligning it - code-wise - to the binding controller.

If it can wait until the latter, we will continue with only the ObjectIdentifier, if not, I can refactor the `SourceRef` out.